triqler/hyperparameters.py

```diff
@@ -44,8 +47,18 @@ def fitPriors(peptQuantRows, params, printImputedVals = False, plot = False):
     #  for mean in means:  
     #    #protGroupDiffs.append(mean - np.mean(means))
     #    protGroupDiffs.append(mean)
-    
+
+
     quantMatrixFiltered = np.log10(np.array([x for x, y in zip(quantMatrix, quantRows) if y.combinedPEP < 1.0]))  
+
+    # Fit prior based on means of missing value
+    if params["missingValuePrior"] == "DIA":
+        for peptide in quantMatrixFiltered:
+            for group in params["groups"]:
+                group_means = peptide[group][~np.isnan(peptide[group])].mean()
+                if group_means != np.nan:
+                    peptide_group_means.append(group_means)
```

Comment:
But now it's not conditional on having at least one missing value, or am I missing something?

Reply:
It simply fits the mean for each group, regardless if they are missing or not, and it can only do this if we have data. I think this is ok?